.github/workflows/scripts/js/common.js

```diff
@@ -0,0 +1,346 @@
+const { ApiPromise, WsProvider } = require("@polkadot/api");
+const { Keyring } = require("@polkadot/api");
+const readline = require("node:readline/promises");
+const { stdin, stdout } = require("node:process");
+
+const rl = readline.createInterface({ input: stdin, output: stdout });
+
+// if keypair is undefined, then dryRun must be true
+async function submitExtrinsic(transaction, keypair, dryRun) {
+  console.log("Submit transaction");
+  if (dryRun) {
+    console.log("Dry run");
+    console.log("Transaction size\n", (transaction.inner.toHex().length - 2) / 2);
+    console.log("Transaction data:");
+    console.log(transaction.inner.toHex());
+
+    if (!dryRun) await multisigTransaction.signAndSend(initiatingSignatory);
+
+    console.log("\n\nTransaction hash:", `0x${Buffer.from(transaction.inner.hash).toString("hex")}`);
+
+    return;
+  }
+
+  await new Promise((resolve, reject) => {
+    transaction.signAndSend(keypair, ({ status, dispatchError }) => {
+      // status would still be set, but in the case of error we can shortcut
+      // to just check it (so an error would indicate InBlock or Finalized)
+      if (dispatchError) {
+        if (dispatchError.isModule) {
+          // for module errors, we have the section indexed, lookup
+          const decoded = api.registry.findMetaError(dispatchError.asModule);
+          const { docs, name, section } = decoded;
+
+          console.log(`${section}.${name}: ${docs.join(" ")}`);
+        } else {
+          // Other, CannotLookup, BadOrigin, no extra info
+          console.log(dispatchError.toString());
+        }
+        reject();
+      }
+
+      if (status.isInBlock) {
+        console.log("Success: transaction in block");
+        resolve();
+      }
+
+      if (status.isFinalized) {
+        console.log("Transaction finalized");
+      }
+    });
+  });
+}
+
+async function democracyProposal(call, type, deposit, submitPreimage, { api, submitterKeypair, dryRun }) {
+  console.log("Preimage data", call.inner.toHex());
+  console.log("Preimage hash", `0x${Buffer.from(call.inner.hash).toString("hex")}`);
+
+  if (submitPreimage) {
+    const submitPreimageTransaction = api.tx.preimage.notePreimage(call.inner.toHex());
+    // await submitExtrinsic(submitPreimageTransaction, submitterKeypair, dryRun);
+  }
```

Comment:
@ebma whose account? Should we bring this up later?

Reply:
@ebma I asked @TorstenStueber and we're going to create different accounts for both pendulum and amplitude. 
This would mean calling this workflow has to be on gitlab, to secure the private key.

Reply:
Sounds good 👍